src/utils/ts-alias.ts

```diff
@@ -0,0 +1,115 @@
+import * as vscode from "vscode";
+import * as path from "path";
+import * as JSON5 from "json5";
+import * as _ from "lodash";
+import * as fse from "fs-extra";
+import { AliasFromTsConfig } from "../options";
+import { WORKSPACE_FOLDER_VARIABLE } from "../constants";
+
+type TsConfigPaths = Record<string, string[]>;
+const cachedMappings = new Map<string, AliasFromTsConfig>();
+
+function memoize(
+  fn: (workfolder: vscode.WorkspaceFolder) => Promise<AliasFromTsConfig>
+) {
+  async function cachedFunction(
+    workfolder?: vscode.WorkspaceFolder
+  ): Promise<AliasFromTsConfig> {
+    if (!workfolder) {
+      return Promise.resolve({});
+    }
+
+    const key = workfolder.name;
+    const cachedMapping = cachedMappings.get(key);
+
+    if (cachedMapping) {
+      return cachedMapping;
+    } else {
+      const result = await fn(workfolder);
+      cachedMappings.set(key, result);
+      return result;
+    }
+  }
+
+  return cachedFunction;
+}
+
+function invalidateCache(workfolder: vscode.WorkspaceFolder) {
+  cachedMappings.delete(workfolder.name);
+}
+
+export function _removePathsSign(paths: TsConfigPaths): TsConfigPaths {
+  const formatPaths: TsConfigPaths = {};
+  function removeEndSign(str: string) {
+    return str.endsWith("*") ? str.slice(0, str.length - 1) : str;
+  }
+
+  Object.keys(paths).forEach((k) => {
+    formatPaths[removeEndSign(k)] = paths[k].map(removeEndSign);
+  });
+
+  return formatPaths;
+}
+
+export function _getAliasFromTsConfigPaths(tsconfig: {
```

Comment:
@iChenLei I don't know whether I cover all the tsconfig paths use cases. By now, I only remove the `* ` if exist in key or value, and join with baseUrl.

Reply:
I think your test case is enough, very comprehensive consideration !
> src/test/suite/ts-alias.test.ts 👍🏻